applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.cpp

```diff
@@ -0,0 +1,144 @@
+//                   
+//                   
+//
+
+
+// Application includes
+#include "utilities/math_utils.h"
+#include "custom_conditions/T_normal_flux_condition.hpp"
+
+namespace Kratos
+{
+
+// ============================================================================================
+// ============================================================================================
+// Default constructor
+template<unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition() : TCondition<TDim, TNumNodes>() {}
+
+// Constructor 1
+template<unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+    : TCondition<TDim, TNumNodes>(NewId, pGeometry) {}
+
+// Constructor 2
+template<unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::TNormalFluxCondition(IndexType NewId, GeometryType::Pointer pGeometry,
+    PropertiesType::Pointer pProperties) : TCondition<TDim, TNumNodes>(NewId, pGeometry, pProperties) {}
+
+// Destructor
+template<unsigned int TDim, unsigned int TNumNodes>
+TNormalFluxCondition<TDim, TNumNodes>::~TNormalFluxCondition() = default;
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TNormalFluxCondition<TDim,TNumNodes>::Create(
+    IndexType NewId,NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(new TNormalFluxCondition(NewId, this->GetGeometry().Create(ThisNodes), pProperties));
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim,TNumNodes>::CalculateRHS(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& CurrentProcessInfo)
+{        
+    //Previous definitions
+    const GeometryType& Geom = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = Geom.IntegrationPoints(this->GetIntegrationMethod());
+    const unsigned int NumGPoints = IntegrationPoints.size();
+    const unsigned int LocalDim = Geom.LocalSpaceDimension();
+    
+    //Containers of variables at all integration points
+    const Matrix& NContainer = Geom.ShapeFunctionsValues(this->GetIntegrationMethod());
+    GeometryType::JacobiansType JContainer(NumGPoints);
+    for(unsigned int i = 0; i < NumGPoints; ++i)
+        (JContainer[i]).resize(TDim,LocalDim,false);
+    Geom.Jacobian( JContainer, this->GetIntegrationMethod());
+    
+    //Condition variables
+    array_1d<double,TNumNodes> NormalFluxVector;
```

Comment:
normal_flux_vector

Reply:
@AlejandroCornejo  Thank you for the review. 
The following I addressed fixes based on your review.